gnocchi/indexer/sqlalchemy.py

```diff
@@ -1071,7 +1073,100 @@ def get_resource(self, resource_type, resource_id, with_metrics=False):
                 q = q.options(sqlalchemy.orm.joinedload(Resource.metrics))
             return session.scalars(q).first()
 
-    def _get_history_result_mapper(self, session, resource_type):
+    def extracts_filters_for_table(self, attribute_filter,
+                                   allowed_keys_for_table=[
+                                       'creator', 'started_at', 'ended_at',
+                                       'user_id', 'project_id',
+                                       'original_resource_id', 'id', 'type']):
+        """Extracts the filters for resource history table.
+
+        Extracts the filters that can be used in the resource history table to
+        apply in the aggregates query that we execute in the database.
+        """
+
+        attribute_filters_to_use = copy.deepcopy(attribute_filter)
+
+        LOG.debug("Executing the processing of attributes filters [%s] for "
+                  "resource history table.", attribute_filters_to_use)
+
+        is_value_list = isinstance(attribute_filters_to_use, list)
+        is_value_dict = isinstance(attribute_filters_to_use, dict)
+        is_value_dict_or_list = (is_value_dict or is_value_list)
+
+        if not is_value_dict_or_list:
+            LOG.debug("Attribute filter [%s] is not of expected types [list "
+                      "or dict]. Therefore, we do not do anything with it.",
+                      attribute_filters_to_use)
+            return attribute_filters_to_use
+
+        if is_value_list:
+            for attribute in attribute_filter:
+                LOG.debug("Sending attribute filter [%s] to be processed, "
+                          "as it is part of a list of attribute filters.",
+                          attribute)
+
+                value_sanitized = self.extracts_filters_for_table(
+                    attribute, allowed_keys_for_table=allowed_keys_for_table)
+
+                if not value_sanitized:
+                    LOG.debug("Value [%s] was totally cleaned after being "
+                              "sanitized. Therefore, we remove it from our "
+                              "attribute filter list.", attribute)
+                    attribute_filters_to_use.remove(attribute)
+                else:
+                    LOG.debug("Replacing value [%s] in list with the sanitized"
+                              "value [%s] in its current position.",
+                              attribute, value_sanitized)
+                    value_index = attribute_filters_to_use.index(attribute)
+                    attribute_filters_to_use[value_index] = value_sanitized
+
+        elif is_value_dict:
+            all_keys = list(attribute_filter.keys())
+            for key in all_keys:
+                value = attribute_filter.get(key)
```

Comment:
```suggestion
            for key, value in attribute_filter.items():
```

Reply:
done.

Reply:
This does not work. Therefore, I reverted:
- https://github.com/gnocchixyz/gnocchi/actions/runs/6278775791/job/17053206922?pr=1307

Reply:
i think it's because you put `for key, value in attribute_filter` instead of `for key, value in attribute_filter.items()`